Source/WebCore/platform/graphics/cg/NativeImageCG.cpp

```diff
@@ -69,6 +70,39 @@ DestinationColorSpace NativeImage::colorSpace() const
     return DestinationColorSpace(CGImageGetColorSpace(m_platformImage.get()));
 }
 
+static bool isHDRNativeImage(NativeImage& image)
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    return CGColorSpaceIsHDR(CGImageGetColorSpace(image.platformImage().get()));
+#pragma clang diagnostic pop
+}
+
+void NativeImage::draw(GraphicsContext& context, const FloatSize& imageSize, const FloatRect& destinationRect, const FloatRect& sourceRect, const ImagePaintingOptions& options)
+{
+    if (sourceRect.isEmpty() || !isHDRNativeImage(*this)) {
+        context.drawNativeImageInternal(*this, imageSize, destinationRect, sourceRect, options);
+        return;
+    }
+
+    // Create a temporary ImageBuffer for destinationRect with the current scaleFator.
```

Comment:
The rest of this function is all about the HDR workaround, so I'd prefer the code be laid out to make that clearer (initially I missed the `! isHDRNativeImage()` early return and was confused). You could put the bulk of this function into a lambda, or another function, with a name that makes it clear that it's an HDR workaround.

There should be a FIXME comment referencing a radar that can tell us when we're able to remove the workaround.

Reply:
A lambda was added because we need to call `GraphicsContext::drawNativeImageInternal()` which is private.

Reply:
rdar://105525195 was filed and a FIXME comment was added.